sh_scrapy/extension.py

```diff
@@ -48,13 +49,24 @@ def spider_closed(self, spider, reason):
         self.pipe_writer.set_outcome(reason)
 
 
+_DOWNLOADER_MIDDLEWARE_WARNING = """\
+sh_scrapy.extension.HubstorageMiddleware functionality is now split between two new middlewares:
+
+- sh_scrapy.middlewares.HubstorageDownloaderMiddleware
+- sh_scrapy.middlewares.HubstorageSpiderMiddleware
+
+Please migrate to new middlewares.
+"""
+
+
 class HubstorageMiddleware(object):
 
     def __init__(self):
+        warnings.warn(_DOWNLOADER_MIDDLEWARE_WARNING)
```

Comment:
I pointed the util because it warns when subclassing, not just when instantiating. It is true the `create_deprecated_class` util was designed for renaming classes but it can be used just to warn about deprecated classes that doesn't have a direct replacement.

```python
from scrapy.utils.deprecate import create_deprecated_class


class HubstorageMiddleware(object):
    pass

HubstorageMiddleware = create_deprecated_class(
    'HubstorageMiddleware',
    HubstorageMiddleware,
    subclass_warn_message="{cls} inherits from deprecated class {old}, "
        "please see ..."
)

print("\nSubclassing HubstorageMiddleware")
class MyExtendedMiddleware(HubstorageMiddleware):
    pass

print("\nInstanciating MyExtendedMiddleware")
MyExtendedMiddleware()
```

output:
```
Subclassing HubstorageMiddleware
demo.py:15: ScrapyDeprecationWarning: __main__.MyExtendedMiddleware inherits from deprecated class __main__.HubstorageMiddleware, please see ... (warning only on first subclass, there may be others)
  class MyExtendedMiddleware(HubstorageMiddleware):

Instanciating MyExtendedMiddleware

```